File: 2022_c#_basics/13-nested-loops/05-travelling.cs
using System;

namespace _05_Travelling
{
    internal class Program
    {
        static void Main(string[] args)
        {
            string destination = Console.ReadLine();

            while (destination != "End")
            {
                double budget = double.Parse(Console.ReadLine());
                double savedMoney = double.Parse(Console.ReadLine());

                while (savedMoney < budget)
                {
                    savedMoney += double.Parse(Console.ReadLine());
                }

                Console.WriteLine($"Going to {destination}!");
                
                destination = Console.ReadLine();
            }
        }
    }
}

File: 2024_java_basics/13-nested-loops/tasks/05-travelling.java
import java.util.Scanner;

public class Travelling {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String destination = scanner.nextLine();

        while (!destination.equals("End")) {
            double budget = scanner.nextDouble();
            double savedMoney = scanner.nextDouble();
            scanner.nextLine(); // Consume the leftover newline character

            while (savedMoney < budget) {
                savedMoney += scanner.nextDouble();
                scanner.nextLine(); // Consume the leftover newline character
            }

            System.out.printf("Going to %s!%n", destination);

            destination = scanner.nextLine();
        }

        scanner.close();
    }
}
